Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -4280,8 +4280,8 @@ class BBQJIT {
                     ? m_jit.branch32(RelationalCondition::Equal, rhsLocation.asGPR(), TrustedImm32(-1))
                     : m_jit.branch64(RelationalCondition::Equal, rhsLocation.asGPR(), TrustedImm64(-1));
                 throwExceptionIf(ExceptionType::IntegerOverflow, isNegativeOne);
-                checkedForNegativeOne = true;
             }
+            checkedForNegativeOne = true;
```

Comment:
Why not remove `checkedForNegativeOne` altogether?  It looks like it's always true whichever path we take.

Reply:
It is false if both are not constant.